src/UI/Components/Diagram/components/FormModal.tsx

```diff
@@ -45,22 +46,17 @@ interface Selected {
 }
 
 const FormModal = ({
-  isOpen,
-  toggleIsOpen,
-  services,
-  cellView,
   onConfirm,
 }: {
-  isOpen: boolean;
-  toggleIsOpen: (value: boolean) => void;
-  services: ServiceModel[];
-  cellView: dia.CellView | null;
   onConfirm: (
     fields: Field[],
     entity: InstanceAttributeModel,
     selected: Selected,
+    cellToEdit: dia.CellView,
```

Comment:
cellToEdit is being used to decide whether the item is in edit/create, but throughout the code it's not clear until the very end where you set the display text. (or not, I'm not sure, the labels of those `words()` confuses me)
How do you know if the user reselects the same cell twice to make a change? Do you keep track of that too? 🤔 

 i'd prefer a logic like in the Form component, where the isEdit is independent of the data of the instance itself, or cell in your case. You'd have the cellData, and the isEdit, or isNew maybe would be more clear.

Reply:
it's temporary, we are moving away from the current modal, I moved as much logic as possible it created some shortcuts but it will be reworked in the next PR